### PR TITLE
Feature: #22470: Set default paths for rct1/rct2 saves on Android

### PR DESCRIPTION
**Pull request by Harry-Hopkinson**
Fix for #22470 - Sets rct1_path and rct2_path to /sdcard/rct1 and /sdcard/rct2 respectively.

I have tested on my android phone and everything appears to work correctly.

### CODE REVIEW

**Comment by Harry-Hopkinson** [comment]
Ah okay, I see what you mean. I will use the Platform.Android.cpp folder

**Comment by Gymnasiast** [comment]
The searchable locations needs a bit of an overhaul anyway, because on Mac and Linux, you’d ideally still want to look in those locations, but in a Wine prefix. For Mac, you’d also want to search in the RCT Classic .app if present, whereas on Linux there are two Steam locations to check (both for the download_depot method and Steam Play). So all in all something to tackle on its own, I would say.

**Comment by tupaschoal** [comment]
Maybe to start with we can declare a function to get these searchable paths on src/openrct2/platform/Platform.h and define if on the platform Android and platform common (this one only when it's not Android). Then we can break the latter into the win32, Unix, Mac versions.

What do you think?

**Comment by tupaschoal** [comment]
If you look closely, we already have made it platform aware a few lines below:

```cpp
        auto steamPath = Platform::GetSteamPath();
        if (!steamPath.empty())
        {
            std::string location = Path::Combine(steamPath, Platform::GetRCT1SteamDir());
            if (RCT1DataPresentAtLocation(location))
            {
                return location;
            }
        }
```

File: src/openrct2/platform/Platform.Common.cpp [hunk 1]
```diff
@@ -203,4 +203,44 @@ namespace OpenRCT2::Platform
         return false;
     }
 
+#ifndef __ANDROID__
```
[the remark this answers is not in the document]
Comment (AaronVanGeffen) [reply src/openrct2/platform/Platform.Common.cpp:206]: I'd rather we do so in this PR as well, to be honest. You can leave an empty stub functions for the other platforms.

**Review by tupaschoal — APPROVED** [review]
Haven't tested, but at a glance it looks alright to me!

**Review by Gymnasiast — COMMENTED** [review]
LGTM2, but want to do a quick test.

**Comment by Gymnasiast** [comment]
Works correctly, needs a changelog entry. Something like:

- Improved: [#22470]: Android: automatically detect RCT2 installs in /sdcard/rct2.

**Comment by Harry-Hopkinson** [comment]
I have added a changelog entry.

**Review by Gymnasiast — APPROVED** [review]

**Comment by CarterMoody** [comment]
What files do I need to dump in my sdcard/rct1/ directory? Mine doesn't seem to be picking it up...

**Pull request merged**